server/sonar-server/src/main/java/org/sonar/server/computation/task/step/TaskInitExtension.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.task.step;
+
+import org.sonar.api.ExtensionPoint;
+import org.sonar.api.ce.ComputeEngineSide;
+
+/**
+ * Extension point that is called during processing of a task
+ * by {@link ExecuteTaskInitExtensionsStep}.
+ */
+@ComputeEngineSide
+@ExtensionPoint
```

Comment:
I'm not sure if the `@ExtensionPoint` annotation is correct here.

implementation of this class should not be returned as extension of the plugin but as classes to be instantiated in the container of the report processing task using a `ReportAnalysisComponentProvider`

This way, instances of implementations of `TaskInitExtension` can be stateful and get injected with components during report processing (such as `CeTask` or whatever)